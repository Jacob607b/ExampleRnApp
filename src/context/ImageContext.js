import { createContext } from "react";

export const ImageContext = createContext({ Images: [null] });

export default ImageContext;
