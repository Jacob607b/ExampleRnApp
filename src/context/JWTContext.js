import { createContext } from "react";

export const JWTContext = createContext({ Jwt: undefined });

export default JWTContext;
