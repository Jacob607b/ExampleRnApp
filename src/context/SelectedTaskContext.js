import { createContext } from "react";

export const SelectedTaskContext = createContext({selectedTask : []});

export default SelectedTaskContext;