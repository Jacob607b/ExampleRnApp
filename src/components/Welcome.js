import React, { useContext, useEffect, useState } from "react";
import {
  View,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  Text,
} from "react-native";
import { TextInput, Card, Button } from "react-native-paper";
import { getAuth, signOut } from "firebase/auth";
// import useFetch from "./functions/useFetch";
import { AuthContext } from "../context/AuthContext";
import { ReadContext } from "../context/ReadContext";
import { IDContext } from "../context/IDContext";
import { JWTContext } from "../context/JWTContext";
import { SelectedTaskContext } from "../context/SelectedTaskContext";
import SelectDropdown from "react-native-select-dropdown";
import { SHADOWS } from "../constants/theme";
const types = ["Electric", "Gas", "Water"];
export default function Welcome({ navigation }) {
  const { readData, setReadData } = useContext(ReadContext);
  const { ID, setID } = useContext(IDContext);
  const { Jwt, setJwt } = useContext(JWTContext);
  const { selectedTask, setSelectedTask } = useContext(SelectedTaskContext);
  // AUTH Firebase User
  const auth = getAuth();
  const user = auth.currentUser;
  const { isLoggedIn, setIsLoggedIn } = useContext(AuthContext);
  const [address, setAddress] = useState(selectedTask?.Unit || "");
  const [needsAnotherInput, setNeedsAnotherInput] = useState(false);
  const [neededInput, setNeededInput] = useState([]);
  const [addedField, setAddedField] = useState([]);

  // runs when user selects a task and when user intially signs in
  useEffect(() => {
    if (selectedTask) {
      setNeedsAnotherInput(true);
      setNeededInput([selectedTask.Type]);
      setAddress(selectedTask.Unit);
      setAddedField([{ type: selectedTask.Type, read: null }]);
      setReadData({
        address: selectedTask.Unit,
        addedField: [{ type: selectedTask.Type, read: null }],
      });
    }
    if (selectedTask.Type == undefined) {
      setNeedsAnotherInput(false);
      setReadData({});
      setAddress("");
    }
  }, [ID, selectedTask]);

  // User Sign out function
  const handleSignOut = async () => {
    try {
      await signOut(auth, user);
      setIsLoggedIn(false);
      // ID context is set to null
      setID(null);
      // JWT is set to undefined
      setJwt(undefined);
      navigation.navigate("Login");
    } catch (error) {
      alert(error.message);
    }
  };
  if (!isLoggedIn) {
    handleSignOut();
  }
  const nagivateToSelectImage = async () => {
    await setReadData({
      ...readData,
      address: address,
      addedField,
    });
    await navigation.navigate("SubmitImage");
    setNeededInput([]);
    setAddedField([]);
    setNeedsAnotherInput(false);
    setAddress("");
  };

  // Called when user manually selects utility to read
  // const handleSelect = (type) => {
  //   setNeedsAnotherInput(true);
  //   setNeededInput([type]);
  //   setAddedField([{ type, read: null }]);
  // };
  // Sets value for Utility field
  const handleAddedFields = (read, index) => {
    setAddedField((prev) => {
      const updatedFields = [...prev];
      updatedFields[index] = { ...updatedFields[index], read };
      setReadData({
        ...readData,
        addedField: updatedFields,
      });
      return updatedFields;
    });
  };
  // resets entire page state and context
  const resetForm = () => {
    setAddedField([]);
    setNeededInput([]);
    setNeedsAnotherInput(false);
    setAddress("");
    setSelectedTask({});
    setReadData({});
  };

  return (
    <>
      <ScrollView style={styles.scrollView}>
        <View style={styles.container}>
          <TextInput
            testID="StreetAddress"
            label="Street Address / Unit"
            value={address}
            style={styles.textInput}
            onChangeText={(e) => setAddress(e)}
          />
          {needsAnotherInput
            ? neededInput.map((needed, i) => {
                return (
                  <React.Fragment key={i}>
                    <View key={i} style={styles.inputContainer}>
                      <Card.Title title={needed} style={styles.type} />
                      <TextInput
                        key={i}
                        name={needed}
                        testID="neededInput"
                        onChangeText={(val) => handleAddedFields(val, i)}
                        style={styles.textInput}
                        keyboardType="numeric"
                        label={`${needed} Read`}
                      />
                    </View>
                  </React.Fragment>
                );
              })
            : null}

          {/* <SelectDropdown
            data={types}
            onSelect={handleSelect}
            testID="SelectUtilty"
            defaultButtonText="Select Utility"
            buttonTextAfterSelection={() => {
              return "Select Utility";
            }}
            style={styles.margin}
          /> */}
          <Text>
            <TouchableOpacity
              testID="reset"
              onPress={resetForm}
              style={styles.margin}
            >
              <Button icon="refresh" compact={true} style={styles.margin}>
                Reset
              </Button>
            </TouchableOpacity>
          </Text>
          <Text>
            <TouchableOpacity
              testID="Next"
              onPress={nagivateToSelectImage}
              style={styles.margin}
            >
              <Button
                icon="arrow-right-bold"
                title="Next"
                compact={true}
                style={styles.margin}
              >
                Next
              </Button>
            </TouchableOpacity>
          </Text>
          <Text>
            <TouchableOpacity
              testID="signOut"
              onPress={handleSignOut}
              style={styles.margin}
            >
              <Button
                icon="account"
                compact={true}
                title="Sign Out"
                style={styles.margin}
              >
                Sign Out
              </Button>
            </TouchableOpacity>
          </Text>
        </View>
      </ScrollView>
    </>
  );
}

const styles = StyleSheet.create({
  scrollView: {
    width: "100%",
    marginTop: 50,
  },
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    shadowColor: SHADOWS.dark,
    shadowRadius: 16.0,
    elevation: 24,
  },
  inputContainer: {
    width: "100%",
    flex: 1,
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
  },
  textInput: {
    borderWidth: 1,
    borderColor: "black",
    width: "80%",
    marginRight: 8,
    padding: 3,
    shadowColor: SHADOWS.dark,
    elevation: 3,
    marginBottom: 5,
  },
  type: {
    borderWidth: 1,
    borderColor: "black",
    width: "80%",
    marginRight: 8,
    padding: 3,
    marginTop: 10,
  },
  margin: {
    marginBottom: 2,
    marginTop: 2,
  },
});
