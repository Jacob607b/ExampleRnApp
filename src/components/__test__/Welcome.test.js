jest.useFakeTimers();
import React from "react";
import Welcome from "../Welcome";
import { IDContext } from "../../context/IDContext";
import { ReadContext } from "../../context/ReadContext";
import { SelectedTaskContext } from "../../context/SelectedTaskContext";
import { JWTContext } from "../../context/JWTContext";
import { AuthContext } from "../../context/AuthContext";
import { fireEvent, render, screen, act } from "@testing-library/react-native";

describe("Welcome Page", () => {
  let useEffect;
  // mockImplemntation to mock useEffect
  const mockUseEffect = () => {
    useEffect.mockImplementation((f) => f());
  };
  useEffect = jest.spyOn(React, "useEffect");
  beforeEach(() => {
    mockUseEffect();
    mockUseEffect();
  });
  it("Renders Inputs from selected task", async () => {
    fetch.mockResponseOnce(
      JSON.stringify({
        data: [
          {
            Unit: "G",
            Type: "ELE",
            Description: "Read Needed",
            completed: false,
          },
        ],
      })
    );
    const setNeededInput = jest.fn();
    jest
      .spyOn(React, "useState")
      .mockImplementationOnce(() => [setNeededInput, () => []]);
    const navigate = jest.fn();
    const setReadData = jest.fn();
    const setSelectedTask = jest.fn();
    const needsAnotherInput = true;
    const setJwt = jest.fn();
    const setIsLoggedIn = jest.fn();
    const mockSelectedTask = {
      Unit: "123",
      Type: "Electric",
    };
    const { getByTestId } = render(
      <AuthContext.Provider value={{ isLoggedIn: true, setIsLoggedIn }}>
        <JWTContext.Provider value={{ Jwt: "1", setJwt }}>
          <IDContext.Provider value={{ ID: "1" }}>
            <ReadContext.Provider value={{ readData: ["1"], setReadData }}>
              <SelectedTaskContext.Provider
                value={{ selectedTask: mockSelectedTask, setSelectedTask }}
              >
                <Welcome
                  navigation={{ navigate }}
                  needsAnotherInput={needsAnotherInput}
                />
              </SelectedTaskContext.Provider>
            </ReadContext.Provider>
          </IDContext.Provider>
        </JWTContext.Provider>
      </AuthContext.Provider>
    );
    const needInput = screen.getByTestId("neededInput");
    expect(needInput).toBeOnTheScreen();
    fireEvent.changeText(needInput, "123");
    const streetAddress = screen.getByTestId("StreetAddress");
    expect(streetAddress).toBeOnTheScreen();
    fireEvent.changeText(streetAddress, "A");
    const resetBtn = getByTestId("reset");
    fireEvent.press(resetBtn);
    const signOut = getByTestId("signOut");
    expect(signOut).toBeOnTheScreen();
    fireEvent.press(signOut);
    await act(() => {
      const nextBtn = getByTestId("Next");
      expect(nextBtn).toBeOnTheScreen();
      fireEvent.press(nextBtn);
    });
    expect(screen).toMatchSnapshot();
  });
  it("Signs out when AuthContext is false", async () => {
    const navigate = jest.fn();
    const setReadData = jest.fn();
    const setSelectedTask = jest.fn();
    const mockSelectedTask = {
      Unit: "123",
      Type: undefined,
    };
    const setJwt = jest.fn();
    const setIsLoggedIn = jest.fn();
    render(
      <AuthContext.Provider value={{ isLoggedIn: false, setIsLoggedIn }}>
        <JWTContext.Provider value={{ setJwt }}>
          <IDContext.Provider value={{ ID: "1" }}>
            <ReadContext.Provider value={{ readData: ["1"], setReadData }}>
              <SelectedTaskContext.Provider
                value={{ selectedTask: mockSelectedTask, setSelectedTask }}
              >
                <Welcome navigation={{ navigate }} />
              </SelectedTaskContext.Provider>
            </ReadContext.Provider>
          </IDContext.Provider>
        </JWTContext.Provider>
      </AuthContext.Provider>
    );
    const needInput = screen.queryByTestId("neededInput");
    expect(needInput).not.toBeOnTheScreen();
  });
  it("Renders with no selected task", async () => {
    const navigate = jest.fn();
    const setReadData = jest.fn();
    const setSelectedTask = jest.fn();
    const mockSelectedTask = {
      Unit: "123",
      Type: undefined,
    };
    const setJwt = jest.fn();
    const setIsLoggedIn = jest.fn();
    render(
      <AuthContext.Provider value={{ isLoggedIn: true, setIsLoggedIn }}>
        <JWTContext.Provider value={{ Jwt: "1", setJwt }}>
          <IDContext.Provider value={{ ID: "1" }}>
            <ReadContext.Provider value={{ readData: ["1"], setReadData }}>
              <SelectedTaskContext.Provider
                value={{ selectedTask: mockSelectedTask, setSelectedTask }}
              >
                <Welcome navigation={{ navigate }} />
              </SelectedTaskContext.Provider>
            </ReadContext.Provider>
          </IDContext.Provider>
        </JWTContext.Provider>
      </AuthContext.Provider>
    );
    const needInput = screen.queryByTestId("neededInput");
    expect(needInput).not.toBeOnTheScreen();
  });
});
