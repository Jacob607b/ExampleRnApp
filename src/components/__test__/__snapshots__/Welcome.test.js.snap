// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Page Renders Inputs from selected task 1`] = `
<RCTScrollView
  style={
    Object {
      "marginTop": 50,
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "elevation": 24,
          "flex": 1,
          "justifyContent": "center",
          "shadowColor": Object {
            "elevation": 14,
            "shadowColor": "#74858C",
            "shadowOffset": Object {
              "height": 7,
              "width": 0,
            },
            "shadowOpacity": 0.41,
            "shadowRadius": 9.11,
          },
          "shadowRadius": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(231, 224, 236, 1)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "borderColor": "black",
              "borderWidth": 1,
              "elevation": 3,
              "marginBottom": 5,
              "marginRight": 8,
              "padding": 3,
              "shadowColor": Object {
                "elevation": 14,
                "shadowColor": "#74858C",
                "shadowOffset": Object {
                  "height": 7,
                  "width": 0,
                },
                "shadowOpacity": 0.41,
                "shadowRadius": 9.11,
              },
              "width": "80%",
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(28, 27, 31, 1)",
              "bottom": 0,
              "height": 1,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
              "zIndex": 1,
            }
          }
          testID="text-input-underline"
        />
        <View
          style={
            Array [
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "minHeight": 56,
              },
            ]
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 4,
                  },
                ],
                "zIndex": 3,
              }
            }
          >
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "opacity": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 30,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -12,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
              testID="StreetAddress-label-active"
            >
              Street Address / Unit
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "opacity": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 30,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -12,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
              testID="StreetAddress-label-inactive"
            >
              Street Address / Unit
            </Text>
          </View>
          <TextInput
            cursorColor="rgba(103, 80, 164, 1)"
            editable={true}
            maxFontSizeMultiplier={1.5}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=" "
            placeholderTextColor="rgba(73, 69, 79, 1)"
            selectionColor="rgba(103, 80, 164, 1)"
            style={
              Array [
                Object {
                  "margin": 0,
                },
                Object {
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 4,
                  "paddingTop": 24,
                },
                Object {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "letterSpacing": 0.15,
                  "lineHeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                Array [
                  Object {},
                ],
                undefined,
              ]
            }
            testID="StreetAddress"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <Text>
        <View
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 2,
              "marginTop": 2,
              "opacity": 1,
            }
          }
          testID="reset"
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 20,
                "marginBottom": 2,
                "marginTop": 2,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": "auto",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "marginLeft": 12,
                          "marginRight": -4,
                        },
                        Object {
                          "marginLeft": 8,
                          "marginRight": 0,
                        },
                        Object {
                          "marginLeft": 6,
                          "marginRight": 0,
                        },
                      ]
                    }
                    testID="button-icon-container"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgba(103, 80, 164, 1)",
                            "fontSize": 18,
                          },
                          Array [
                            Object {
                              "lineHeight": 18,
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "material-community",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰑐
                    </Text>
                  </View>
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Array [
                            Object {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            Object {
                              "marginHorizontal": 16,
                            },
                            Object {
                              "marginHorizontal": 8,
                            },
                            false,
                            Object {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Reset
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Text>
      <Text>
        <View
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 2,
              "marginTop": 2,
              "opacity": 1,
            }
          }
          testID="Next"
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 20,
                "marginBottom": 2,
                "marginTop": 2,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": "auto",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container"
              title="Next"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "marginLeft": 12,
                          "marginRight": -4,
                        },
                        Object {
                          "marginLeft": 8,
                          "marginRight": 0,
                        },
                        Object {
                          "marginLeft": 6,
                          "marginRight": 0,
                        },
                      ]
                    }
                    testID="button-icon-container"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgba(103, 80, 164, 1)",
                            "fontSize": 18,
                          },
                          Array [
                            Object {
                              "lineHeight": 18,
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "material-community",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰜴
                    </Text>
                  </View>
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Array [
                            Object {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            Object {
                              "marginHorizontal": 16,
                            },
                            Object {
                              "marginHorizontal": 8,
                            },
                            false,
                            Object {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Next
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Text>
      <Text>
        <View
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 2,
              "marginTop": 2,
              "opacity": 1,
            }
          }
          testID="signOut"
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 20,
                "marginBottom": 2,
                "marginTop": 2,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": "auto",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container"
              title="Sign Out"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "marginLeft": 12,
                          "marginRight": -4,
                        },
                        Object {
                          "marginLeft": 8,
                          "marginRight": 0,
                        },
                        Object {
                          "marginLeft": 6,
                          "marginRight": 0,
                        },
                      ]
                    }
                    testID="button-icon-container"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgba(103, 80, 164, 1)",
                            "fontSize": 18,
                          },
                          Array [
                            Object {
                              "lineHeight": 18,
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "material-community",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰀄
                    </Text>
                  </View>
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Array [
                            Object {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            Object {
                              "marginHorizontal": 16,
                            },
                            Object {
                              "marginHorizontal": 8,
                            },
                            false,
                            Object {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Sign Out
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
